scripts/cmake/sdk_android.cmake

```diff
@@ -0,0 +1,134 @@
+    elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+        if(DEFINED ENV{LOCALAPPDATA})
+            _defold_android_sdk_add_toolchains("$ENV{LOCALAPPDATA}/Android/Sdk")
+        endif()
+        if(DEFINED ENV{APPDATA})
+            _defold_android_sdk_add_toolchains("$ENV{APPDATA}/Android/Sdk")
+        endif()
+    endif()
+endif()
+
+# Pick the most recent toolchain file
+if(_ANDROID_TOOLCHAINS)
+    list(SORT _ANDROID_TOOLCHAINS COMPARE NATURAL)
+    list(REVERSE _ANDROID_TOOLCHAINS)
+    list(GET _ANDROID_TOOLCHAINS 0 _ANDROID_TOOLCHAIN_FILE)
+endif()
+
+if(EXISTS "${_ANDROID_TOOLCHAIN_FILE}")
+    get_filename_component(_toolchain_dir "${_ANDROID_TOOLCHAIN_FILE}" DIRECTORY)
+    # Typical path: <ndk_root>/build/cmake/android.toolchain.cmake
+    # NDK root is either parent (android.toolchain.cmake in root) or grandparent (build/cmake/...)
+    get_filename_component(_ndk_parent "${_toolchain_dir}" DIRECTORY)
+    if(EXISTS "${_ndk_parent}/source.properties")
+        set(_ANDROID_NDK_ROOT "${_ndk_parent}")
+    else()
+        get_filename_component(_ndk_grandparent "${_ndk_parent}" DIRECTORY)
+        set(_ANDROID_NDK_ROOT "${_ndk_grandparent}")
+    endif()
+
+    if(_SDKS_DIR AND _ANDROID_TOOLCHAIN_FILE MATCHES "^${_SDKS_DIR}.*")
+    defold_log("sdk_android: Using packaged Android NDK: ${_ANDROID_NDK_ROOT}")
+    else()
+    defold_log("sdk_android: Using local Android SDK NDK: ${_ANDROID_NDK_ROOT}")
+    endif()
+    defold_log("sdk_android: Toolchain file: ${_ANDROID_TOOLCHAIN_FILE}")
+
+    if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+        set(CMAKE_TOOLCHAIN_FILE "${_ANDROID_TOOLCHAIN_FILE}" CACHE FILEPATH "Android NDK toolchain file" FORCE)
+    endif()
+    set(ANDROID_NDK "${_ANDROID_NDK_ROOT}" CACHE PATH "Android NDK root" FORCE)
+else()
+    message(FATAL_ERROR "sdk_android: Failed to find android.toolchain.cmake in packaged or local Android SDKs")
+endif()
+
+
+if(TARGET_PLATFORM MATCHES "arm64-android")
+    # For arm64-android, ensure Clang uses aarch64-linux-android21 target triple
+    set(ANDROID_ABI "arm64-v8a" CACHE STRING "Android ABI" FORCE)
+    set(ANDROID_NATIVE_API_LEVEL 21 CACHE STRING "Android API Level" FORCE)
+    set(ANDROID_TOOLCHAIN "aarch64-linux-android${ANDROID_NATIVE_API_LEVEL}-clang" CACHE STRING "Android Toolchain" FORCE)
+
+else(TARGET_PLATFORM MATCHES "armv7-android")
+    # For armv7-android, ensure Clang uses armv7-linux-android21 target triple
+    set(ANDROID_ABI "armeabi-v7a" CACHE STRING "Android ABI" FORCE)
+    set(ANDROID_NATIVE_API_LEVEL 19 CACHE STRING "Android API Level" FORCE)
+    # Use the canonical Clang triplet for 32-bit ARM
+    set(ANDROID_TOOLCHAIN "armv7a-linux-androideabi${ANDROID_NATIVE_API_LEVEL}-clang" CACHE STRING "Android Toolchain" FORCE)
+
+endif()
+
+defold_log("ANDROID_ABI: ${ANDROID_ABI}")
+defold_log("ANDROID_NATIVE_API_LEVEL: ${ANDROID_NATIVE_API_LEVEL}")
+defold_log("ANDROID_TOOLCHAIN: ${ANDROID_TOOLCHAIN}")
```

Comment:
As before, when running `./scripts/build.py check_sdk --platform=arm64-android --verbose` you'll get an output like this:
```bash
-- sdk.cmake:
-- sdk_android.cmake:
-- sdk_android: Using packaged Android NDK: /Users/mathiaswesterdahl/work/defold/tmp/dynamo_home/ext/SDKs/android-ndk-r25b
-- sdk_android: Toolchain file: /Users/mathiaswesterdahl/work/defold/tmp/dynamo_home/ext/SDKs/android-ndk-r25b/build/cmake/android.toolchain.cmake
-- ANDROID_ABI: arm64-v8a
-- ANDROID_NATIVE_API_LEVEL: 21
-- ANDROID_TOOLCHAIN: aarch64-linux-android21-clang
-- check_install: Environment looks OK for arm64-android
```